src/sec_security_asn1kc.c

```diff
@@ -761,11 +760,20 @@ Sec_Asn1KC *SecAsn1KC_Decode(SEC_BYTE *buf, SEC_SIZE buf_len)
     const unsigned char *c_buf = buf;
     Sec_Asn1KC *ret = NULL;
 
-    if (buf_len > LONG_MAX)
+    if (UINT32_MAX > LONG_MAX)
+    {
+        if (buf_len > LONG_MAX)
+        {
+             SEC_LOG_ERROR("buf length rollover");
+             return NULL;
+        }
+    }
```

Comment:
Not sure why this is needed.  SEC_SIZE is typedefed to uint32_t.  Why compare to LONG_MAX here?